pixeldata/src/gdcm.rs

```diff
@@ -0,0 +1,257 @@
+        let rows = rows(self)?;
+
+        let photometric_interpretation = photometric_interpretation(self)?;
+        let pi_type = GDCMPhotometricInterpretation::from_str(&photometric_interpretation)
+            .map_err(|_| Error::UnsupportedPhotometricInterpretation {
+                pi: photometric_interpretation.clone(),
+            })?;
+
+        let transfer_syntax = &self.meta().transfer_syntax;
+        let registry =
+            TransferSyntaxRegistry
+                .get(&&transfer_syntax)
+                .context(UnsupportedTransferSyntax {
+                    ts: transfer_syntax,
+                })?;
+        let ts_type = GDCMTransferSyntax::from_str(&registry.uid()).map_err(|_| {
+            Error::UnsupportedTransferSyntax {
+                ts: transfer_syntax.clone(),
+            }
+        })?;
+
+        let samples_per_pixel = samples_per_pixel(self)?;
+        let bits_allocated = bits_allocated(self)?;
+        let bits_stored = bits_stored(self)?;
+        let high_bit = high_bit(self)?;
+        let pixel_representation = pixel_representation(self)?;
+        let rescale_intercept = rescale_intercept(self);
+        let rescale_slope = rescale_slope(self);
+
+        let decoded_pixel_data = match pixel_data.value() {
+            Value::PixelSequence {
+                fragments,
+                offset_table: _,
+            } => {
+                if fragments.len() > 1 {
+                    // Bundle fragments and decode multi-frame dicoms
+                    UnsupportedMultiFrame.fail()?
+                }
+                let decoded_frame = decode_single_frame_compressed(
+                    &fragments[0],
+                    cols.into(),
+                    rows.into(),
+                    pi_type,
+                    ts_type,
+                    samples_per_pixel,
+                    bits_allocated,
+                    bits_stored,
+                    high_bit,
+                    pixel_representation,
+                )
+                .map_err(|_| Error::UnknownGdcmError)?;
+                decoded_frame.to_vec()
+            }
+            Value::Primitive(p) => {
+                // Non-encoded, just return the pixel data
+                p.to_bytes().to_vec()
+            }
+            Value::Sequence { items: _, size: _ } => InvalidPixelData.fail()?,
+        };
+
+        Ok(DecodedPixelData {
+            data: Cow::from(decoded_pixel_data),
+            cols: cols.into(),
+            rows: rows.into(),
+            photometric_interpretation,
+            samples_per_pixel,
+            bits_allocated,
+            bits_stored,
+            high_bit,
+            pixel_representation,
+            rescale_intercept,
+            rescale_slope,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use dicom_object::open_file;
+    use dicom_test_files;
+    use rstest::rstest;
+    use std::path::Path;
+
+    #[cfg(feature = "gdcm")]
```

Comment:
Was this needed? The module is only checked with the feature anyway.

Reply:
No you are right. This can be removed.